x-pack/metricbeat/module/azure/billing/service.go

```diff
@@ -46,49 +52,76 @@ func NewService(config azure.Config) (*UsageService, error) {
 		return nil, err
 	}
 
-	forecastsClient := consumption.NewForecastsClientWithBaseURI(config.ResourceManagerEndpoint, config.SubscriptionId)
 	usageDetailsClient := consumption.NewUsageDetailsClientWithBaseURI(config.ResourceManagerEndpoint, config.SubscriptionId)
+	forecastsClient := costmanagement.NewForecastClientWithBaseURI(config.ResourceManagerEndpoint, config.SubscriptionId)
 
-	forecastsClient.Authorizer = authorizer
 	usageDetailsClient.Authorizer = authorizer
+	forecastsClient.Authorizer = authorizer
 
 	service := UsageService{
 		usageDetailsClient: &usageDetailsClient,
-		forecastsClient:    &forecastsClient,
+		forecastClient:     &forecastsClient,
 		context:            context.Background(),
 		log:                logp.NewLogger("azure billing service"),
 	}
 
 	return &service, nil
 }
 
-// GetForecast fetches the forecast for the given filter.
-func (service *UsageService) GetForecast(filter string) ([]consumption.Forecast, error) {
-	response, err := service.forecastsClient.List(service.context, filter)
+// GetForecast fetches the forecast for the given scope and time interval.
```

Comment:
also a big fan of this level of detail, especially when dealing with these fairly opaque APIs

Reply:
Yeah, it's for the sanity of my current and future selves, and any other person will have to deal with this.